cmr-exchange/ses-plugin/src/cmr/sizing/formats.clj

```diff
@@ -192,18 +234,46 @@
                                 avg-compression-rate))
               (+ total-estimate
                  (* total-granule-input-bytes
-                    avg-compression-rate
+                    (or avg-compression-rate 0) 
                     (/ (/ total-granule-input-bytes granule-count) avg-gran-size)))))
           0
           variables))
 
 (defn estimate-size
-  [fmt granule-count vars granule-metadata-size params]
-  (case (keyword (string/lower-case fmt))
-    :dods (estimate-dods-size granule-count vars params)
-    :nc (estimate-netcdf3-size granule-count vars granule-metadata-size params)
-    :nc4 (estimate-netcdf4-size granule-count vars params)
-    :ascii (estimate-ascii-size granule-count vars params)
-    (do
-      (log/errorf "Cannot estimate size for %s (not implemented)." fmt)
-      {:errors [(str "Cannot estimate size for format: " fmt)]})))
+  [svcs granule-count vars granule-metadata-size params]
+  (let [svc-id (:service-id params)
+        svc-type (if svc-id
+                   ;; There is at most one svc associated with svc-id 
+                   (get-in (first svcs) [:umm :Type])              
+                   opendap-service-type)
+        svc-type-lc (when svc-type
+                      (string/lower-case svc-type))
+        format (:format params)
+        fmt-lc (if format 
+                 (string/lower-case format)
+                 default-format)
+        ses-fmt (get-in service-type->formats-mapping [svc-type-lc fmt-lc])]
+    (if (or (= svc-type-lc opendap-service-type)
+            (= svc-type-lc egi-service-type))
+      (case (keyword ses-fmt)
+        :dods (estimate-dods-size granule-count vars params)
+        :nc (estimate-netcdf3-size granule-count vars granule-metadata-size params)
+        :nc4 
+          (estimate-netcdf4-shapefile-size 
+            granule-count vars params (get ses-formats->compression-format-mapping ses-fmt))
+        :ascii (estimate-ascii-size granule-count vars params)
+        :shapefile 
+          (estimate-netcdf4-shapefile-size 
+            granule-count vars params (get ses-formats->compression-format-mapping ses-fmt))
+        :tabular_ascii {:errors ["tabular_ascii not implemented"]}
```

Comment:
These should be human friendly error messages being returned. Make them a sentence like the ones below.

Reply:
Done.